### PR TITLE
Disable memory tracking by default in the Launcher and Editor

### PR DESCRIPTION
**Pull request by burelc-amzn**
This disables the memory tracking by default in the Editor and the Launcher. Tracking memory in those scenarios is too expensive to be on by default.

### CODE REVIEW

**Review by lawsonamzn — APPROVED** [review]

**Comment by lawsonamzn** [comment]
LGTM - I'd definitely recommend breakpoint testing though, I remember last time I looked in here disabling them was harder than expected since some things used to inherit defaults not directly connected to the create struct.

File: Code/Editor/EditorToolsApplication.cpp [hunk 1]
```diff
@@ -105,7 +105,17 @@ namespace EditorInternal
...
         // Must be done before creating QApplication, otherwise asserts when we alloc
-        AzToolsFramework::ToolsApplication::Start({}, params);
+        AzToolsFramework::ToolsApplication::Start({
+                /* .m_useExistingAllocator = */ false,
+                /* .m_allocationRecordsSaveNames = */ false,
+                /* .m_allocationRecordsAttemptDecodeImmediately = */ false,
+                /* .m_autoIntegrityCheck = */ false,
+                /* .m_markUnallocatedMemory = */ false,
+                /* .m_doNotUsePools = */ false,
+                /* .m_enableScriptReflection = */ false,
```
Comment (mbalfour-amzn) [inline Code/Editor/EditorToolsApplication.cpp:115]: Should "enableScriptReflection" be enabled?

Comment (burelc-amzn) [reply Code/Editor/EditorToolsApplication.cpp]: Fixed.

**Review by mbalfour-amzn — APPROVED** [review]

**Review by spham-amzn — APPROVED** [review]

**Review by cgalvan — APPROVED** [review]

**Comment by gadams3** [comment]
We also need to make these changes to other tools, especially material canvas and shader management console. I believe this is affecting material canvas load times because graph canvas does a lot of inefficient container copying and reallocating while creating graph UI and applying style sheets. If it needs to be done in that many places should the default settings change?

https://github.com/o3de/o3de/blob/development/Gems/Atom/Tools/MaterialCanvas/Code/Source/main.cpp
https://github.com/o3de/o3de/blob/development/Gems/Atom/Tools/MaterialCanvas/Code/Source/main.cpp
https://github.com/o3de/o3de/blob/development/Gems/Atom/Tools/PassCanvas/Code/Source/main.cpp
https://github.com/o3de/o3de/blob/development/Gems/Atom/Tools/ShaderManagementConsole/Code/Source/main.cpp

**Comment by lawsonamzn** [comment]
It may be easier to change the systemallocator.cpp and osallocator.cpp's constructors to not auto-enable profiling ?

**Comment by gadams3** [comment]
> It may be easier to change the systemallocator.cpp and osallocator.cpp's constructors to not auto-enable profiling ? That would then affect all projects.

That's my recommendation. Disable it by default. Disabling the setting locally resulted in a ~4X improvement in the cases I was testing. But it also identified things that need to be optimized in graph canvas that shouldn't be excessively cloning containers of smart pointers while applying style sheets on graph objects.

**Comment by lawsonamzn** [comment]
> > It may be easier to change the systemallocator.cpp and osallocator.cpp's constructors to not auto-enable profiling ? That would then affect all projects.
> 
> That's my recommendation. Disable it by default. Disabling the setting locally resulted in a ~4X improvement in the cases I was testing. But it also identified things that need to be optimized in graph canvas that shouldn't be excessively cloning containers of smart pointers while applying style sheets on graph objects.

yes, I just mean that its a smaller change to just cut those lines out
https://github.com/o3de/o3de/blob/560e148d6bd5dc221cf6598f3ed033d7a1c4a0fb/Code/Framework/AzCore/AzCore/Memory/SystemAllocator.cpp#L32
and the same kind of thing in OSAllocator, than the original change proposed here.

Then only those would need to change.

**Comment by lawsonamzn** [comment]
Did not mean to close :)

**Review by SergeyAMZN — APPROVED** [review]

**Comment by burelc-amzn** [comment]
I'll open a new PR for the revised approach, based on the suggestions here to disable tracking in the OSAllocator and SystemAllocator by default.

**Pull request closed without merging**